content/chainguard/chainguard-images/tooling/building-java-containers-with-jib.md

```diff
@@ -0,0 +1,588 @@
+[INFO]
+[INFO] Container entrypoint set to [java, -cp, @/app/jib-classpath-file, com.example.demo.DemoApplication]
+[INFO] Container entrypoint set to [java, -cp, @/app/jib-classpath-file, com.example.demo.DemoApplication]
+[INFO]
+[INFO] Built image to Docker daemon as linky
+...
+```
+
+
+This command has compiled the application, built the container, and loaded it
+into the local Docker instance.
+
+To verify that the image was built:
+
+```shell
+docker images linky
+```
+
+Which will provide details on the new image:
+
+```
+REPOSITORY   TAG       IMAGE ID       CREATED         SIZE
+linky        latest    43171844f68e   3 minutes ago   465MB
+```
+
+You should also scan the image for CVEs. This example uses
+[grype](https://github.com/anchore/grype), but you can use your preferred
+scanner: 
+
+```
+grype linky
+```
+
+This should give you output similar to the following:
+
+```
+✔ Vulnerability DB                [updated]
+ ✔ Loaded image                                                                 linky:latest
+ ✔ Parsed image                    sha256:078478c4d49bae684d8337c3a97df00722f7d1c0d8507e6e1ae
+ ✔ Cataloged contents              a7123bf3bfc2296155ab226593f9f0d79b322c29da5d94af90fdc267cc
+   ├── ✔ Packages                        [75 packages]
+   ├── ✔ File metadata                   [1,391 locations]
+   ├── ✔ Executables                     [124 executables]
+   └── ✔ File digests                    [1,391 files]
+ ✔ Scanned for vulnerabilities     [0 vulnerability matches]
+   ├── by severity: 0 critical, 0 high, 0 medium, 0 low, 0 negligible
+No vulnerabilities found
+```
+
+In this case, there were no CVEs found. Try comparing those results to images built with different base images.
+
+## Stage 4 — Testing the Containerized Application
+
+After building your container image, test it to ensure it works correctly.
+Start by running the container locally:
+
+```shell
+docker run -p 8080:8080 linky
+```
+
+The application will start, and you'll see Spring Boot's startup logs. In a
+separate terminal, test the endpoints:
+
+```shell
+curl http://localhost:8080/
+```
+
+This should return:
+
+```
+Hello from Jib and Chainguard!
+```
+
+Test the health endpoint:
+
+```shell
+curl http://localhost:8080/health
+```
+
+Expected output:
+
+```
+Application is healthy
+```
+
+Stop the container by pressing `Ctrl+C` in the terminal where it's running.
+
+## Stage 5 — Building and pushing to a remote registry
+
+One of the strongest points of jib is that you don't need Docker installed to
+build and distribute container images. In this step we will change `pom.xml` to
+push the image directly to a container registry.
+
+This example uses `ttl.sh`, which is a free-to-use Docker registry for short-lived hosting of images. You can also change the code to use your own registry such as a local one or the Docker Hub.
+
+Replace the previous plugin definition with this one (or add this one if you skipped step 3):
+
+
+```xml
+ <plugin>
```

Comment:
I liked the way the previous one was formatted with more spaces before `<plugin>` so I could just paste it into the xml file